openquake/server/views.py

```diff
@@ -318,12 +319,18 @@ def calc(request, id=None):
                            allowed_users, user['acl_on'], id)
 
     response_data = []
-    for hc_id, owner, status, calculation_mode, is_running, desc in calc_data:
+    for hc_id, owner, status, calculation_mode, is_running, desc, pid \
+            in calc_data:
         url = urlparse.urljoin(base_url, 'v1/calc/%d' % hc_id)
+        abortable = False
+        if bool(is_running):
+            if psutil.Process(pid).username() == owner:
```

Comment:
`psutil.Process(pid)` can raise an error `psutil.NoSuchProcess` if the process has terminated, the exception must be trapped and managed.